meilisearch/gatsby-plugin-meilisearch
# Test document indexation

Closes #8 

## What's inside this PR ? 

- Validate function is now a module (in order to be imported in the tests). The watch mode has been updated accordingly
- Addition of jest as dependency, it's configuration file, and allowing of `.env` file
- Addition of a file `validate-options.test.js` testing the validation of the options
- Addition of a file `index-to-meilisearch.test.js` testing the logic of the plugin
- Update of the scripts inside the `package.json`
- Update of `CONTRIBUTING.md` for further explaining of the project's testing
- Update of the CI

## How to test ?

You should have a MeiliSearch and the front's graphQL server running in order to run the tests.
You can do so with the following commands: 

```bash
docker run -it --env MEILI_MASTER_KEY=masterKey -p 7700:7700 getmeili/meilisearch:v0.21.0rc6
```

Then, you can run the tests:

`yarn test`

or: 

`yarn test:watch`

## Screenshots

<img width="683" alt="Capture d’écran 2021-08-02 à 10 25 28" src="https://user-images.githubusercontent.com/30866152/127829220-d4930aa5-72e0-4d83-aad1-f19c12506e89.png">
